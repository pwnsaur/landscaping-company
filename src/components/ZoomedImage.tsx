import Image from 'next/image';
import styled from 'styled-components';

interface ZoomedImageProps {
  src: string;
  alt: string;
  width: number;
  height: number;
  close: () => void;
}

const ZoomedImage = ({ src, alt, width, height, close }: ZoomedImageProps) => {
  const aspectRatio = height / width;

  return (
    <Overlay onClick={close}>
      <ImageWrapper $aspectRatio={aspectRatio}>
        <StyledImage src={src} alt={alt} width={width} height={height} />
      </ImageWrapper>
    </Overlay>
  );
};

export default ZoomedImage;

const StyledImage = styled(Image)`
  width: 100%;
  height: auto;
`;

const ImageWrapper = styled.div<{ $aspectRatio: number }>`
  position: relative;
  max-width: 80%;
  max-height: 80%;
  padding-bottom: calc(80% * ${({ $aspectRatio }) => $aspectRatio});
`;

const Overlay = styled.div`
  position: fixed;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  background-color: rgba(255, 255, 255, 0.8);
  display: flex;
  align-items: center;
  justify-content: center;
  padding: 5rem;
  z-index: 6;
`;
