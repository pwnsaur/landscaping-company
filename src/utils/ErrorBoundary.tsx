import React, { ErrorInfo } from 'react';

interface Props {
  children: React.ReactNode;
}

interface State {
  hasError: boolean;
}

class ErrorBoundary extends React.Component<Props, State> {
  constructor(props: Props) {
    super(props);

    this.state = { hasError: false };
  }

  static getDerivedStateFromError(_: Error): State {
    return { hasError: true };
  }

  componentDidCatch(error: Error, errorInfo: ErrorInfo) {
    console.log({ error, errorInfo });
  }

  render() {
    if (this.state.hasError) {
      return (
        <div>
          <h1>Lapu neizdevās ielādēt.</h1>
        </div>
      );
    }

    return this.props.children;
  }
}

export default ErrorBoundary;
