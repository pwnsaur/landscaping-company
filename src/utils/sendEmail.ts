import axios, { AxiosResponse } from 'axios';

import { FormData } from '@/types/contactForm';

type ServerResponse = {
  messageId: string;
  response: string;
};

type SendEmailFn = (
  formData: FormData
) => Promise<AxiosResponse<ServerResponse>>;

const sendEmail: SendEmailFn = async (formData: FormData) => {
  return axios({
    method: 'post',
    url: '/api/send-mail',
    data: formData,
  });
};

export default sendEmail;
