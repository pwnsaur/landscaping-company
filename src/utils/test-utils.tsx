import { ReactElement, ReactNode } from 'react';
import { ThemeProvider } from 'styled-components';

import { theme } from '@/styles/theme';
import useIsMobile from '@/utils/hooks/useIsMobile';
import { render, RenderOptions } from '@testing-library/react';

interface TestWrapperProps {
  children: ReactNode;
}

const TestWrapper = ({ children }: TestWrapperProps) => {
  const isMobile = useIsMobile();

  return (
    <ThemeProvider theme={{ ...theme, isMobile }}>{children}</ThemeProvider>
  );
};

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, 'queries'>
) => {
  return render(ui, { wrapper: TestWrapper, ...options });
};

// eslint-disable-next-line import/export
export * from '@testing-library/react';

// eslint-disable-next-line import/export
export { customRender as render };
