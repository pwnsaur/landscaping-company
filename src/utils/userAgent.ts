export function isMobileUserAgent(userAgent: string): boolean {
  return /Android|iPhone|iPod|BlackBerry|IEMobile|Opera Mini/i.test(userAgent);
}
