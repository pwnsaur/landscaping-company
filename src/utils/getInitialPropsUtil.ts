import { AppContext, AppInitialProps } from 'next/app';

import { isMobileUserAgent } from './userAgent';

export async function getInitialPropsUtil(
  appContext: AppContext
): Promise<AppInitialProps & { isMobile: boolean }> {
  let pageProps = {};

  if (appContext.Component.getInitialProps) {
    pageProps = await appContext.Component.getInitialProps(appContext.ctx);
  }

  const userAgent = appContext.ctx.req?.headers['user-agent'] || 'rekt';
  const isMobile = isMobileUserAgent(userAgent);

  return { pageProps, isMobile };
}
