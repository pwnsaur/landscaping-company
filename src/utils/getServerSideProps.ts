import { GetServerSideProps } from 'next';

import { isMobileUserAgent } from '@utils/userAgent';

export const getServerSideProps: GetServerSideProps = async ({ req }) => {
  const userAgent = req.headers['user-agent'] || 'about rekt';
  const isMobile = isMobileUserAgent(userAgent);
  return { props: { isMobile } };
};
