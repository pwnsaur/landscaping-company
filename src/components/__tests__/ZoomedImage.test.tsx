import ZoomedImage from '@/components/ZoomedImage';
import { render } from '@/utils/test-utils';

describe('ZoomedImage component', () => {
  test('matches the snapshot', () => {
    const { asFragment } = render(
      <ZoomedImage
        src={'https://meow.com'}
        alt={'meow'}
        width={0}
        height={0}
        close={function (): void {
          throw new Error('Function not implemented.');
        }}
        previous={function (): void {
          throw new Error('Function not implemented.');
        }}
        next={function (): void {
          throw new Error('Function not implemented.');
        }}
        zoomedImageIndex={0}
        imagesLength={0}
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
