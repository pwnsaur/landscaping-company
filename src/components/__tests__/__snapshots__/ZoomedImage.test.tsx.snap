// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ZoomedImage component matches the snapshot 1`] = `
<DocumentFragment>
  .c2 {
  width: 100%;
  height: auto;
}

.c1 {
  position: relative;
  max-width: 80%;
  max-height: 80%;
  padding-bottom: calc(80% * NaN);
}

.c0 {
  position: fixed;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  background-color: rgba(255,255,255,0.8);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 5rem;
  z-index: 6;
}

<div
    class="c0"
  >
    <div
      class="c1"
    >
      <img
        alt="meow"
        class="c2"
        data-nimg="1"
        decoding="async"
        height="0"
        loading="lazy"
        src="/_next/image?url=https%3A%2F%2Fmeow.com&w=16&q=75"
        srcset="/_next/image?url=https%3A%2F%2Fmeow.com&w=16&q=75 1x"
        style="color: transparent;"
        width="0"
      />
    </div>
  </div>
</DocumentFragment>
`;
